src/cider/exec/operator/join/HashTableFactory.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#pragma once
+
+#include <map>
+#include <string>
+
+#include "exec/operator/join/BaseHashTable.h"
+#include "exec/operator/join/CiderLinearProbingHashTable.h"
+
+namespace cider_hashtable {
+
+// To be added
+// This enum is used for cider internal only
+// Outside cider will need to define their own enum
+enum HashTableType { LINEAR_PROBING, CHAINED, CK_INT8 };
+
+struct MurmurHash {
+  size_t operator()(int64_t rawHash) {
+    rawHash ^= unsigned(rawHash) >> 33;
+    rawHash *= 0xff51afd7ed558ccdL;
+    rawHash ^= unsigned(rawHash) >> 33;
+    rawHash *= 0xc4ceb9fe1a85ec53L;
+    rawHash ^= unsigned(rawHash) >> 33;
+    return rawHash;
+  }
+};
+
+struct Equal {
+  bool operator()(int lhs, int rhs) { return lhs == rhs; }
+};
+
+template <typename Key,
+          typename Value,
+          typename Hash,
+          typename KeyEqual,
+          typename Grower,
+          typename Allocator>
+class HashTableSelector {
+ public:
+  // TODO: Define hashtable selection strategy in here
+  HashTableType getHashTableTypeForJoin(std::string data_type,
+                                        int initial_size,
+                                        double ratio) {}
+
+  template <typename... Args>
+  std::unique_ptr<BaseHashTable<Key, Value, Hash, KeyEqual, Grower, Allocator>>
+  createForJoin(HashTableType hashtable_type, Args&&... args);
```

Comment:
Maybe we should return a HashTable in `getHashTableTypeForJoin` and remove `createForJoin` or rename `createForJoin` to `createHashTable`? If users got `hashtable_type` which means the type of HashTable they should use, they only need a `create` function no matter it is Join or Aggregation.